### PR TITLE
Fixed the hooks and backported some changes

### PR DESCRIPTION
**Pull request by mildbyte**

### CODE REVIEW

**Comment by jamesblackburn** [comment]
CC: @richardbounds @Wilfred @slavi

File: arctic/auth.py [hunk 1]
```diff
@@ -25,4 +26,8 @@ def get_auth(host, app_name, database_name):
     """
     Authentication hook to allow plugging in custom authentication credential providers
     """
-    return None
+    return _get_auth_hook(host, app_name, database_name)
+
+def register_get_auth_hook(hook):
```
Comment (jamesblackburn) [inline arctic/auth.py]: Move this to ```hooks.py```

File: arctic/store/version_store.py [hunk 2]
```diff
@@ -273,7 +273,7 @@ def _write_handler(self, version, symbol, data, **kwargs):
             handler = self._bson_handler
         return handler
 
-    def read(self, symbol, as_of=None, from_version=None, **kwargs):
+    def read(self, symbol, as_of=None, from_version=None, allow_secondary=None, **kwargs):
```
Comment (jamesblackburn) [inline arctic/store/version_store.py]: Rebase so this is in a separate commit?

File: arctic/tickstore/tickstore.py [hunk 3]
```diff
@@ -325,24 +325,35 @@ def _set_or_promote_dtype(self, column_dtypes, c, dtype):
                 dtype = np.dtype('f8')
             column_dtypes[c] = np.promote_types(column_dtypes.get(c, dtype), dtype)
 
-    def _prepend_image(self, document, im):
+    def _prepend_image(self, document, im, rtn_length, column_dtypes, column_set, columns):
```
Comment (jamesblackburn) [inline arctic/tickstore/tickstore.py]: Are there tests for this change?

**Comment by Wilfred** [comment]
LGTM :+1:

**Comment by jamesblackburn** [comment]
LGTM, thanks!

**Pull request merged**